lib/src/css_mirroring/rulesets_processor.dart

```diff
@@ -0,0 +1,82 @@
+part of scissors.src.css_mirroring.bidi_css_generator;
+
+abstract class NodeProcessor {
```

Comment:
Processor is as vague as Do, Run, etc.

Seems like the common trait of all these is they remove stuff.
Also, not sure you're ever using that base class.

Looks like each of these classes is also just conceptually a function, that takes a TextEditTransaction (and other things) and returns a PendingRemovals.